design-guidelines/form-ux.md

```diff
@@ -0,0 +1,189 @@
+# Form Design Guidelines
+
+A form may be used to collect data that is required to complete a task.
+
+## Form contexts
+
+The context in which a form is rendered depends on the complexity of the form and how critical the form's purpose is to the rest of the app.
+
+### Full page
+
+A form may be rendered on it's own page helps users to focus on completing their task with minimal distractions in the UI.
+
+Render a form on it's own page if the form:
+
+- is complex and has many fields
+- will be used to complete an action that is very important
+
+### Section of a page
+
+A form may be rendered on a page alongside additional content to avoid taking users out of their current flow.
+
+Render a form alongside other content if:
+
+- the user may need to reference page content to help them fill out the form correctly
+- the form is not so complex that it deserves it's own page
+- it would be too disruptive to take a user away from the current page
+- the page has content that is related to the form
+
+### Dialog modal
+
+A form may be rendered in a dialog modal to keep the user on their current page while directing their focus to a form. The first input of the form should receive focus when the modal is opened.
+
+Render a form in a dialog modal if:
+
+- there are very few fields in the form
+- there is no logical place to render the form on the current page
+- the user won't need to reference content that will be covered by the dialog
+
+### Inline inputs
+
+In lieu of rendering an entire form, a single input can be used in place of the data it controls. Inputs would replace the data once the user has explicity toggled edit mode.
```

Comment:
I like that. We don't have an example for that in dcos-ui, do we?

Have we done this in kommander?

Reply:
We do this in Dispatch. I _think_ I've seen this pattern on one of the dcos-ui settings pages.

Reply:
> is no longer in vogue 

yeah, i remember doing this in 2013 XD

my question was addressed here, thanks!